src/MongoDB.Driver/Encryption/LibMongoCryptController.cs

```diff
@@ -0,0 +1,681 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using MongoDB.Bson.IO;
+using MongoDB.Bson.Serialization.Serializers;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.WireProtocol;
+using MongoDB.Libmongocrypt;
+
+namespace MongoDB.Driver.Encryption
+{
+    internal sealed class LibMongoCryptController : IBinaryDocumentFieldDecryptor, IBinaryCommandFieldEncryptor, IDisposable
+    {
+        #region static
+        private static bool AcceptAnyCertificate(
+            object sender,
+            X509Certificate certificate,
+            X509Chain chain,
+            SslPolicyErrors sslPolicyErrors)
+        {
+            return true;
```

Comment:
Just a reminder that we need to remove this before shipping

Reply:
why? 
NOTE: We use this logic in some other places like: 
https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver/ClusterRegistry.cs#L179
https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilderExtensions.cs#L176

Reply:
Shouldn't we connect to `mongocryptd` with the same security settings  that the user supplies to us for MongoClient? It seems dangerous to default to NOT checking SSL certificates.

In the snippets you linked, it seems we only skip verification if `AllowSecureTls` is true
https://github.com/mongodb/mongo-csharp-driver/blob/87d19f6382f77215e8d749849b8e17434cbed4f4/src/MongoDB.Driver/ClusterRegistry.cs#L134-L136
https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilderExtensions.cs#L97